### PR TITLE
[WIP][FrontendBundle]: Changed core_shop_frontend.pimcore_admin.install scalar nodes to arra…

### PR DESCRIPTION
**Pull request by khusseini**
…y nodes

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #497 

Changed structure of `CoreShopFrontendBundle` Configuration of `pimcore_admin.install` to use array nodes instead of scalar nodes

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
@khusseini Can you check other bundles as well? They all have different configurations for installation stuff.

**Comment by dpfaffenbauer** [comment]
@khusseini also travis doesn't seem to like your changes ;), can you check again pls

File: src/CoreShop/Bundle/AddressBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -229,7 +229,11 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                     ->arrayNode('install')
                         ->addDefaultsIfNotSet()
                         ->children()
-                            ->scalarNode('admin_translations')->defaultValue(['@CoreShopAddressBundle/Resources/install/pimcore/admin-translations.yml'])->end()
+                            ->arrayNode('admin_translations')
+                                ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/AddressBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/CustomerBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -142,7 +142,11 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                     ->arrayNode('install')
                         ->addDefaultsIfNotSet()
                         ->children()
-                            ->scalarNode('admin_translations')->defaultValue(['@CoreShopCustomerBundle/Resources/install/pimcore/admin-translations.yml'])->end()
+                            ->arrayNode('admin_translations')
+                                ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/CustomerBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -91,15 +91,29 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
...
                 ->children()
-                    ->arrayNode('install')
-                        ->addDefaultsIfNotSet()
-                        ->children()
-                            ->scalarNode('routes')->defaultValue(['@CoreShopFrontendBundle/Resources/install/pimcore/staticroutes.yml'])->end()
-                            ->scalarNode('documents')->defaultValue(['@CoreShopFrontendBundle/Resources/install/pimcore/documents.yml'])->end()
-                            ->scalarNode('image_thumbnails')->defaultValue(['@CoreShopFrontendBundle/Resources/install/pimcore/image-thumbnails.yml'])->end()
-                            ->scalarNode('translations')->defaultValue(['@CoreShopFrontendBundle/Resources/install/pimcore/translations.yml'])->end()
+                    ->arrayNode('routes')
+                        ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -91,15 +91,29 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
...
+                            ->defaultValue(['@CoreShopFrontendBundle/Resources/install/pimcore/staticroutes.yml'])
+                        ->end()
+                    ->end()
+                    ->arrayNode('documents')
+                        ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -91,15 +91,29 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
...
+                            ->defaultValue(['@CoreShopFrontendBundle/Resources/install/pimcore/documents.yml'])
+                        ->end()
+                    ->end()
+                    ->arrayNode('image_thumbnails')
+                        ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php [hunk 6]
```diff
@@ -91,15 +91,29 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
...
+                            ->defaultValue(['@CoreShopFrontendBundle/Resources/install/pimcore/image-thumbnails.yml'])
+                        ->end()
+                    ->end()
+                    ->arrayNode('translations')
+                        ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/FrontendBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/OrderBundle/DependencyInjection/Configuration.php [hunk 7]
```diff
@@ -477,8 +477,16 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                     ->arrayNode('install')
                         ->addDefaultsIfNotSet()
                         ->children()
-                            ->scalarNode('admin_translations')->defaultValue(['@CoreShopOrderBundle/Resources/install/pimcore/admin-translations.yml'])->end()
-                            ->scalarNode('grid_config')->defaultValue(['@CoreShopOrderBundle/Resources/install/pimcore/grid-config.yml'])->end()
+                            ->arrayNode('admin_translations')
+                                ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/OrderBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/OrderBundle/DependencyInjection/Configuration.php [hunk 8]
```diff
@@ -477,8 +477,16 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
...
+                                    ->defaultValue(['@CoreShopOrderBundle/Resources/install/pimcore/admin-translations.yml'])
+                                ->end()
+                            ->end()
+                            ->arrayNode('grid_config')
+                                ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/OrderBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/PaymentBundle/DependencyInjection/Configuration.php [hunk 9]
```diff
@@ -151,9 +151,11 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                             ->scalarNode('payment')->defaultValue('/bundles/coreshoppayment/pimcore/css/payment.css')->end()
                         ->end()
                     ->end()
-                    ->scalarNode('permissions')
-                        ->cannotBeOverwritten()
-                        ->defaultValue(['payment_provider'])
+                    ->arrayNode('permissions')
+                        ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/PaymentBundle/DependencyInjection/Configuration.php]: don't overwrite permission configurations, you cannot change them anyway and its not part of this PR

File: src/CoreShop/Bundle/ProductBundle/DependencyInjection/Configuration.php [hunk 10]
```diff
@@ -228,14 +228,20 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
...
+                            ->cannotBeOverwritten()
+                            ->defaultValue(['product_price_rule'])
+                        ->end()
                     ->end()
                     ->arrayNode('install')
                         ->addDefaultsIfNotSet()
                         ->children()
-                            ->scalarNode('admin_translations')->defaultValue(['@CoreShopProductBundle/Resources/install/pimcore/admin-translations.yml'])->end()
+                            ->arrayNode('admin_translations')
+                                ->scalarPrototype()
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/ProductBundle/DependencyInjection/Configuration.php]: works like this:

```
->treatNullLike(array())
->scalarPrototype()->end()
->defaultValue([''])
```

File: src/CoreShop/Bundle/ProductBundle/DependencyInjection/Configuration.php [hunk 11]
```diff
@@ -228,14 +228,20 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                             ->scalarNode('product')->defaultValue('/bundles/coreshopproduct/pimcore/css/product.css')->end()
                         ->end()
                     ->end()
-                    ->scalarNode('permissions')
-                        ->cannotBeOverwritten()
-                        ->defaultValue(['product_price_rule'])
+                    ->arrayNode('permissions')
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/ProductBundle/DependencyInjection/Configuration.php]: don't overwrite permission configurations, you cannot change them anyway and its not part of this PR

File: src/CoreShop/Bundle/ShippingBundle/DependencyInjection/Configuration.php [hunk 12]
```diff
@@ -190,9 +190,11 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                             ->scalarNode('shipping')->defaultValue('/bundles/coreshopshipping/pimcore/css/shipping.css')->end()
                         ->end()
                     ->end()
-                    ->scalarNode('permissions')
-                        ->cannotBeOverwritten()
-                        ->defaultValue(['carrier', 'shipping_rule'])
+                    ->arrayNode('permissions')
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/ShippingBundle/DependencyInjection/Configuration.php]: don't overwrite permission configurations, you cannot change them anyway and its not part of this PR

File: src/CoreShop/Bundle/StoreBundle/DependencyInjection/Configuration.php [hunk 13]
```diff
@@ -107,9 +107,11 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                             ->scalarNode('store_item')->defaultValue('/bundles/coreshopstore/pimcore/css/store.css')->end()
                         ->end()
                     ->end()
-                    ->scalarNode('permissions')
-                        ->cannotBeOverwritten()
-                        ->defaultValue(['store'])
+                    ->arrayNode('permissions')
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/StoreBundle/DependencyInjection/Configuration.php]: don't overwrite permission configurations, you cannot change them anyway and its not part of this PR

File: src/CoreShop/Bundle/TaxationBundle/DependencyInjection/Configuration.php [hunk 14]
```diff
@@ -195,9 +195,11 @@ private function addPimcoreResourcesSection(ArrayNodeDefinition $node)
                             ->scalarNode('tax_item')->defaultValue('/bundles/coreshoptaxation/pimcore/css/taxation.css')->end()
                         ->end()
                     ->end()
-                    ->scalarNode('permissions')
-                        ->cannotBeOverwritten()
-                        ->defaultValue(['tax_rate', 'tax_rule_group'])
+                    ->arrayNode('permissions')
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/TaxationBundle/DependencyInjection/Configuration.php]: don't overwrite permission configurations, you cannot change them anyway and its not part of this PR

**Comment by khusseini** [comment]
Hey there, just wanted to let you know that I won't be able to work on it for another week

**Comment by dpfaffenbauer** [comment]
closed due to https://github.com/coreshop/CoreShop/pull/529
